packages/native-stack/src/types.tsx

```diff
@@ -452,6 +454,43 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   gestureResponseDistance?: ScreenProps['gestureResponseDistance'];
+  /**
+   * Changes the gesture for dismissing the screen to the previous one.
+   *
+   * Supported values:
+   * - "swipeRight": swipe right to dismiss the screen
+   * - "swipeLeft": swipe left to dismiss the screen
+   * - "swipeUp": swipe up to dismiss the screen
+   * - "swipeDown": swipe down to dismiss the screen
+   * - "verticalSwipe": swipe up or down to dismiss the screen
+   * - "horizontalSwipe": swipe left or right to dismiss the screen
+   * - "twoDimensionalSwipe": swipe in any direction to dismiss the screen
+   *
+   * Only supported on iOS and Android.
+   */
+  goBackGesture?: GoBackGesture;
+  /**
+   * Changes the animation for dismissing the screen when making a swipe gesture.
+   * You can choose one of the built-in transition presets, by using `ScreenTransition` from `react-native-reanimated`.
+   *
+   * Supported values:
+   * - ScreenTransition.SwipeRight
+   * - ScreenTransition.SwipeLeft
+   * - ScreenTransition.SwipeUp
+   * - ScreenTransition.SwipeDown
+   * - ScreenTransition.Horizontal
+   * - ScreenTransition.Vertical
+   * - ScreenTransition.TwoDimensional
+   * - ScreenTransition.SwipeRightFade
+   *
+   * You can also construct custom screen transition by providing an object with `topScreenFrame` and `belowTopScreenFrame` properties.
+   * Only supported on iOS and Android.
+   */
+  transitionAnimation?: AnimatedScreenTransition;
```

Comment:
Similar comment as above, for animation related options, we use the `animation` prefix. So it'd be better to match the convention.

I'm also curious how this interacts with the `animation` option. Does it override it?

Reply:
This prop does not override it, but rather it chooses the animation while doing the gesture to go back, while animation is only for classic transitioning - by clicking on the element (or performing navigation actions) that navigates to given screen.

Reply:
Maybe `animationForGesture`. The `Back` seems redundant since all gesture related options are for back gesture.

Reply:
Done in https://github.com/react-navigation/react-navigation/pull/11943/commits/301cd8dace25c46ed8b0f5931333176b07dcc99c.